cdap-ai-ext-vertexai/src/main/java/io/cdap/cdap/ai/ext/vertexai/VertexAIProvider.java

```diff
@@ -54,17 +55,19 @@ public void initialize(AIProviderContext context) throws Exception {
   }
 
   /**
-   * Summarizes the application details in the specified format using Vertex AI. Currently, this
-   * method is not implemented and returning a null string.
+   * Summarizes the application details in the specified format using Vertex AI.
    *
    * @param applicationDetail Details of the application to be summarized.
    * @param format The format in which the summary should be provided.
-   * @return This method currently returning a null string.
+   * @return This method will be returning a string which is summary in required format.
    */
   @Override
   public String summarizeApp(ApplicationDetail applicationDetail, String format) {
     try {
-      GenerateContentResponse response = model.generateContent(conf.getPrompt().getPipelineMarkdownSummary());
+      Gson gson = new Gson();
+      String pipelineDetail = gson.toJson(applicationDetail);
+      String payload = pipelineDetail + "\n" + conf.getPrompt().getPipelineMarkdownSummary();
```

Comment:
Create List<Content> instead of concatenating strings.

Reply:
Used StringBuilder as per discussion.